File: src/main/scala/com/celestia/models/Camera.scala

package com.celestia.models

import com.celestia.interfaces.ICamera
import graphicslib3D.Matrix3D

/**
 * Class for encapsulating the camera view
 * as a mutatable object
 */
case class Camera(
  override val x:Float,
  override val y:Float,
  override val z:Float
) extends ICamera


//  override val position:Matrix3D,
//  override val rotation:Matrix3D,
//  override val zoom:Matrix3D

File: src/main/java/com/celestia/models/Camera.java

package com.celestia.models;

public class Camera implements ICamera {
    public final float x; 
    public final float y; 
    public final float z; 
    
    public Camera(float x, float y, float z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }
}

